### PR TITLE
Enable unit tests and fix all the issues found thanks to them

### PR DESCRIPTION
**Pull request by mardy**
It might be simpler to review this commit by commit, as there are many unrelated changes. The unifying point is that all these commits are either related to enabling running unit tests, or to fixing errors found thanks to them.

### CODE REVIEW

**Review by hummlbach — COMMENTED** [review]
I looked over the commits for half an hour, but i would need half an hour per commit (due to me lacking knowledge about qtpim, eds and the way unit tests are done here, inspite the changes being very small). So I concentrate on the two changes in the production code and pose a question on of them...

File: organizer/qorganizer-eds-saverequestdata.cpp
Comment (hummlbach) [inline organizer/qorganizer-eds-saverequestdata.cpp:41]: I'm not sure whether I agree on that comment...
After the call of `nextSourceId` at qorganizer-eds-engine.cpp:567 it is checked `sourceId` is null, but also if `data->end()` is true too `data->finish()` gets called.
But on the other hand `data->finish()` gets called at qorganizer-eds-engine.cpp:163 if `sourceId.isEmpty()`. Is that really consistent and correct?
Should the comment refer to qorganizer-eds-engine.cpp:585, where the default collection is assumed when the `sourceId` is empty? Could we theoretically change it to `isNull` there instead of setting it to `""` here?

Comment (mardy) [reply organizer/qorganizer-eds-saverequestdata.cpp:41]: Thanks @hummlbach for the review! Yes, I agree that it's confusing, and I've added a comment at qorganizer-eds-engine.cpp:585, as you suggested. I think the other comment I've added before is still very important, so I've left it too.
The reason why in qorganizer-eds-engine.cpp:163 we use a different check is because in that situation we are *reading* data out of the databse, not *writing* it: and we know that all items stored into the database will have a non-null, non-empty collection ID assigned to them.

Comment (hummlbach) [reply organizer/qorganizer-eds-saverequestdata.cpp:41]: I still don't get it completely... But don't mind :grin:

**Comment by hummlbach** [comment]
Something else: I was about to click on the merge button, when I saw the collections test was failing for amd64 with the last build... To really make sense of the tests, we should fix that right...?

**Comment by mardy** [comment]
You are right, and I'll probably remove that line from the test, since I don't understand what it does and the comment next to it makes it even more confusing. :-)
But I notice that there's another warning from another test case, and I'll definitely fix that for good :-)

**Comment by mardy** [comment]
BTW, once I've fixed it, please just approve the MP, don't merge it: I prefer when the author merges the MP himself, to keep the responsibilities more evident (at least when the author has merge rights).

**Comment by mardy** [comment]
@hummlbach, I've updated the branch. Unfortunately, after many hours of debugging, I couldn't get around the unpredictability of the "readonly" property, so the tests are still randomly failing. But I think it's still good if we run them (I've modified the `debian/rules` file to ignore their exit status), and [I've filed a bug](https://gitlab.gnome.org/GNOME/evolution-data-server/issues/53) in the EDS project.

**Review by hummlbach — APPROVED** [review]

**Pull request merged**
